distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/AbstractContentFunction.java

```diff
@@ -0,0 +1,93 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.xebialabs.restito.semantics.Function;
+
+/**
+ * Boiler-plate processing logic for the mock csw server response function.
+ */
+public abstract class AbstractContentFunction implements Function<Response, Response> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractContentFunction.class);
+
+    protected char[] responseMessage;
+
+    protected HeaderCapture headerCapture;
+
+    public AbstractContentFunction(String responseMessage, HeaderCapture headerCapture) {
+        this.responseMessage = responseMessage.toCharArray();
+        this.headerCapture = headerCapture;
+    }
+
+    /**
+     * Implementation of the Function interface's apply method. This class can also be used as a
+     * Function<Response, Response> directly in the Restito response if custom actions are needed.
+     *
+     * @param response Response object correlating to the incoming request. Used to write data
+     *                 back to the requesting client.
+     * @return Response New state of the response object correlating to the incoming request.
+     */
+    @Override
+    public Response apply(Response response) {
+        return respond(response);
+    }
+
+    /**
+     * Sends message to the client one character at a time. Appropriate headers must be set on
+     * the Response object before calling this method.
+     *
+     * @param response  Response object containing an output stream to the client
+     * @param byteRange Object containing the range of bytes to send to the client
+     */
+    protected abstract Response send(Response response, ByteRange byteRange);
+
+    /**
+     * Default response logic for all restito csw mocked responses.
+     */
+    private Response respond(Response response) {
+        Map<String, String> requestHeaders = Collections.emptyMap();
+        if (headerCapture != null) {
+            requestHeaders = headerCapture.getHeaders();
+            LOGGER.debug("ChunkedContentResponse: extracted request headers [{}]", requestHeaders);
+        }
+
+        //  If range header is present, return 206 - Partial Content status
+        // Set Content-Range header if byte offset is specified
+        ByteRange byteRange;
+        if (StringUtils.isNotBlank(requestHeaders.get("range"))) {
+            byteRange = new ByteRange(requestHeaders.get("range"),
+                    responseMessage.length,
+                    responseMessage);
+            response.setStatus(HttpStatus.PARTIAL_CONTENT_206);
+            response.setHeader("Content-Range", byteRange.contentRangeValue());
+            LOGGER.debug("ChunkedContentResponse: Response range header set to [Content-Range: {}]",
+                    byteRange.contentRangeValue());
+        } else {
+            response.setStatus(HttpStatus.OK_200);
+            byteRange = new ByteRange(0, responseMessage.length - 1, responseMessage);
```

Comment:
The inconsistency between the two ctors bothers me. In one, you provide start/end values necessitating this `-1`; in the other, you provide range values and total size so don't decrement the size. It reads oddly to me.
I'd prefer if both ctors took the total size and your `ByteRange` logic managed the fenceposts.

As I see that these are the only two lines, at least in the current code, that are creating `ByteRange`s, you could manage that simply. In fact, since these are the only creations, you could probably go with something like:
`ByteRange(String rangeHeader, int totalSize, String message)`
and
`ByteRange(int totalSize, String message)`
as your constructors. You seem to always go from 0 to size-1 in the no-header case anyway.